fairscale/nn/misc/grad_bucket.py

```diff
@@ -0,0 +1,148 @@
+from typing import Any, Callable, List, Optional, Union
+
```

Comment:
the idea behind that was to simplify sharded_ddp, which is already complicated enough. The reasoning is the same as before, just with the torch.futures and not workhandle